tests/src/test/java/org/eclipse/hono/tests/client/ClientTestBase.java

```diff
@@ -232,8 +233,15 @@ public void testSendingMessages(final TestContext ctx) throws Exception {
             setup.complete();
         }));
 
-        sender.setDispositionHandler((id, disposition) -> accepted.countDown());
+        sender.setDispositionHandler((id, disposition) -> {
+            accepted.countDown();
+            //TODO temp fix to the test, until we have a logic to resend released qos1 messages
```

Comment:
If by

> until we have a logic to resend released qos1 messages

you mean that Hono Server should try to re-send the message(s) then I disagree. IMHO we should simply propagate the disposition upstream to the original sender and let the sender decide what he wants to do (re-send or discard). Otherwise we are at risk to introduce a *message backlog* within Hono, effectively taking responsibility for the message(s) (which we do not want to do).